src/miner.cpp

```diff
@@ -566,6 +562,10 @@ bool CreateRestOfTheBlock(CBlock &block, CBlockIndex* pindexPrev,
                                 mrc_map.insert(make_pair(*mrc_cpid, make_pair(tx.GetHash(), mrc)));
                             }
                         } //TryCpid()
+                    } else {
+                        // The output limit is reached, skip MRC transactions as their
+                        // fee rate is not correct anymore.
+                        goto end;
```

Comment:
This is one of the few times I agree with a goto statement. :) This is a big change in behavior. Before, the transactions would be accepted, but not included in the mrc tx map (i.e. paid in the claim). Now we are going to ignore the ones "off of the end", and then delete them when they are stale, which will be right after this block gets staked. I actually like the new behavior better. It is enabled once we went down this road of fee boosting capability.

Reply:
(See my other comment about adding the staleness check to the ConnectBlock routine.)